src/store/modules/accounts/hdWallet.js

```diff
@@ -35,21 +36,25 @@ export default {
 
       throw new Error('Unsupported protocol');
     },
-    async confirmTxSigning({ dispatch }, { txBase64, app }) {
+    // Confirm the transaction with the modal window. Not usable from the background
+    // as the background should use popups.
+    // TODO replace with `checkOrAskPermission` taken from the permissions composable
+    async confirmTxSigning(_, { txBase64, app }) {
       const { openModal } = useModals();
+      const { checkTransactionSignLimit } = usePermissions();
 
       if (!isTxOfASupportedType(txBase64)) {
         await openModal(MODAL_CONFIRM_RAW_SIGN, { data: txBase64, app });
         return;
       }
       const txObject = unpackTx(txBase64);
 
-      const checkTransactionSignPermission = await dispatch('permissions/checkTransactionSignPermission', {
-        ...txObject,
-        host: app?.host || null,
-      }, { root: true });
+      const hasPermission = checkTransactionSignLimit(
+        app?.host || null,
+        txObject, // TODO verify if this is correct
```

Comment:
TODO shall be verified before merging this.

Reply:
You are the the author of this code :)
I think this is wrong, but I'm afraid to touch it as it actually works. What do you think about creating separate task for this?

Reply:
For now, I agree to keep it as it is. We can add a related task to the board. 

I am repeating this @Liubov-crypto this PR must be tested for all types of transaction signing. Full regression testing.